src/pages/postgraphile/jwk-verification.md

```diff
@@ -201,6 +201,29 @@ appropriate and succinct response:
 }
 ```
 
+If, however, you prefer to assign a default PostgreSQL role when no token has been provided (in a way replicating `pgDefaultRole` behavior) then that's also possible by intercepting when `err.code === 'credentials_required'`:
+
+```javascript
+const authErrors = (err, req, res, next) => {
+  if (err.name === "UnauthorizedError") {
+    console.log(err); // You will still want to log the error...
+    if (err.code === 'credentials_required') {
+      // No token provided, so assign a default PostgreSQL role (see pgDefaultRole) and continue
+      req.user = {
+        role: 'unauthenticated'
+      }
+      next()
+    } else {
+      // Some other auth problem, send back as normal
+      res.status(err.status).json({ errors: [{ message: err.message }] });
```

Comment:
Should we check err.status is a valid status code? What should it fall back to - 403?

Reply:
Not sure of the internals of `express-jwt` (if you always get a valid `status`)... but falling-back to `403` sounds a very good idea!

Reply:
The combination of `credentialsRequired: false` and:

```
pgSettings: req => {
  const settings = {}
  if (req.user && req.user.role) {
    settings.role = req.user.role
  } else {
    settings.role = 'unauthenticated'
  }
  return settings
}
```

...works great for me, thanks for your help @benjie !

Reply:
(and because of that, like you say, there's no need to set `req.user` manually, which must be better!)